### PR TITLE
[#930] Pluggable build mechanism

### PR DESCRIPTION
**Pull request by mcdoker18**
* Added new legion-mlflow deployment
* Added Makefile(cluster creation/deletion, linters, export hiera script, build images and so on)
* Removed statsd exporter. Currently metrics are saved into mlflow service
* Added scripts to import hiera configuration in one step
* Replace nexus to google docker registry for model docker repository
* Added little doc about local development

### CODE REVIEW

File: terraform/modules/gcp/gke_cluster/main.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 provider "google-beta" {
-  version = "2.9"
+  version = "~> 2.12"
```
Comment (kirillmakhonin) [inline terraform/modules/gcp/gke_cluster/main.tf]: Why it is not exact version?

File: terraform/modules/legion/main.tf [hunk 2]
```diff
@@ -24,28 +24,89 @@ provider "aws" {
 }
 
 ########################################################
-# Install Legion dependencies
+# TLS keys
 ########################################################
-data "helm_repository" "legion" {
-  name = "legion"
-  url  = var.legion_helm_repo
+data "aws_s3_bucket_object" "tls-secret-key" {
```
Comment (kirillmakhonin) [inline terraform/modules/legion/main.tf:29]: Do we use AWS even for GCP deployment for storing TLS certificates?

File: terraform/modules/legion/templates/mlflow.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  enabled: true
+  tlsEnabled: true
+  tlsSecretName: "${cluster_name}-tls"
+
+security:
+  enabled: false
+
+tracking_server:
+  ingress:
+    domain: "mlflow.${cluster_name}.${root_domain}"
```
Comment (kirillmakhonin) [inline terraform/modules/legion/templates/mlflow.yaml]: Why do we need .ingress.globalDomain with .tracking_server.ingress.domain? If we do not specify .tracking_server.ingress.domain template with .ingress.globalDomain should be used.

File: .gitwhitelisted [hunk 4]
```diff
@@ -45,3 +45,4 @@ username : unicode
 86.57.255.92/32
 # Pipeline hiera import
 token: ":pkcs7_
+password: |
```
Comment (kirillmakhonin) [inline .gitwhitelisted:48]: Looks like thing that is not secure (real password can be on a next line)

**Review by kirillmakhonin — CHANGES_REQUESTED** [review]

File: terraform/modules/legion/main.tf (hunk 2 above)
Comment (mcdoker18) [reply terraform/modules/legion/main.tf:29]: Yes

File: terraform/modules/gcp/gke_cluster/main.tf (hunk 1 above)
Comment (mcdoker18) [reply terraform/modules/gcp/gke_cluster/main.tf]: I reverted this change.

File: .gitwhitelisted (hunk 4 above)
Comment (mcdoker18) [reply .gitwhitelisted:48]: Provide an alternative

File: terraform/modules/legion/templates/mlflow.yaml (hunk 3 above)
Comment (mcdoker18) [reply terraform/modules/legion/templates/mlflow.yaml]: Fixed

File: .gitwhitelisted (hunk 4 above)
Comment (kirillmakhonin) [reply .gitwhitelisted:48]: two-line regex?

Comment (mcdoker18) [reply .gitwhitelisted:48]: Can you provide an example?

**Review by maxkochubey — APPROVED** [review]

File: terraform/modules/gcp/gke_cluster/variables.tf [hunk 5]
```diff
@@ -183,15 +133,95 @@ variable "gke_node_tag" {
...
-# variable "ip_range_pods" {
-#   default = "10.1.1.0/24"
-#   description = "The secondary ip range to use for pods"
-# }
-
-# variable "ip_range_services" {
-#   default = "10.1.2.0/24"
-#   description = "The secondary ip range to use for pods"
-# }
+variable "main_nodel_pool" {
```
Comment (alexander-semenets) [inline terraform/modules/gcp/gke_cluster/variables.tf]: mistype - nodel

Comment (mcdoker18) [reply terraform/modules/gcp/gke_cluster/variables.tf]: Fixed

File: .gitwhitelisted (hunk 4 above)
Comment (mcdoker18) [reply .gitwhitelisted:48]: Sorry, but I do not understand how to fix this problem. If you provide an example, then I'll apply it.

**Review by alexander-semenets — APPROVED** [review]

**Review by karbyshevds — APPROVED** [review]

**Review by alexander-semenets — APPROVED** [review]

**Pull request merged**